march_data_collector/src/march_data_collector/cp_calculator.py

```diff
@@ -0,0 +1,54 @@
+from math import sqrt
+
+import rospy
+from visualization_msgs.msg import Marker
+
+
+class CPCalculator(object):
+
+    def __init__(self, com_mark, tf_buffer, foot_link):
+        self.tf_buffer = tf_buffer
+        self.foot_link = foot_link
+        self.prev_x = com_mark.pose.position.x
+        self.prev_y = com_mark.pose.position.y
+        self.prev_t = com_mark.header.stamp
+
+        self.marker = Marker()
+
+        self.marker.header.frame_id = 'world'
+        self.marker.type = self.marker.SPHERE
+        self.marker.action = self.marker.ADD
+        self.marker.pose.orientation.w = 1.0
+        self.marker.color.a = 1.0
+        self.marker.color.g = 1.0
+        self.marker.scale.x = 0.03
+        self.marker.scale.y = 0.03
+        self.marker.scale.z = 0.03
+
+        g = 9.81  # gravity constant
+        z_zero = 0.715500019353  # height of CoM
+        self.MULTIPLICATION_CONSTANT = sqrt(z_zero / g)
+
+    def calculate_cp(self, com_mark):
+        current_time = com_mark.header.stamp
+        time_difference = (current_time - self.prev_t).to_sec()
+        if current_time is not self.prev_t:
+            x_dot = (com_mark.pose.position.x - self.prev_x) / time_difference
+            y_dot = (com_mark.pose.position.y - self.prev_y) / time_difference
+
+            trans = self.tf_buffer.lookup_transform('world', self.foot_link, rospy.Time())
```

Comment:
The default `rospy.Time()`[ constructor initializes on 0 seconds](https://docs.ros.org/kinetic/api/rospy/html/rospy.rostime.Time-class.html#__init__), which might not be what you want.
```suggestion
            trans = self.tf_buffer.lookup_transform('world', self.foot_link, rospy.get_rostime())
```

Reply:
By setting the time argument to zero, it will get the latest and that is exactly what we want.
http://docs.ros.org/indigo/api/tf2_ros/html/c++/classtf2__ros_1_1Buffer.html#acabbd72cae8f49fb3b6ede3be7e34c55